Add local method

jayrbolton opened the pull request:
This adds a simple way to directly call lineage_wf, exposing all options. Added docs for using the local method in README.md and a simple test in the existing test class

I made my own util in lib/kb_Msuite/Utils/simple_run_checkm.py because the existing util files were overly convoluted for what I was trying to do

jayrbolton commented:
Requesting review from @Tianhao-Gu  if you have time.

Tianhao-Gu commented on lib/kb_Msuite/kb_MsuiteImpl.py:
```diff
@@ -128,6 +129,49 @@ def run_checkM_lineage_wf(self, ctx, params):
+        out_dir = params['output_dir']
+        log_path = params['log_path']
+        run_checkm(in_dir, out_dir, log_path, params.get('options'))
+        result = {}
```
Should run_checkm return a stdout string? Same string should be returned in lineage_wf result.

jayrbolton replied:
It writes all stdout and stderr into a file in log_path. In several cases I found that stdout is going to be extremely long, thousands of lines, so I think people might not want to have it in a string / in memory by default.

The downside is it's not user friendly to have the user open up a log file to see any errors. But also it's not easy for us to automatically know if there was an error -- checkm uses stdout for many errors and does not use error codes.

Tianhao-Gu replied:
👍  Could you change spec file to reflect the method is returning an empty dict?

Tianhao-Gu commented on lib/kb_Msuite/Utils/simple_run_checkm.py:
```diff
@@ -0,0 +1,24 @@
+      options - optional - dictionary of lineage_wf options
+    """
+    args = ['checkm', 'lineage_wf', input_dir, output_dir]
+    for opt, val in (options or {}).iteritems():
```
should we check legitimacy of options? Or stdout already gives pretty informative messages for bad options?

Tianhao-Gu replied:
👍

jayrbolton commented on lib/kb_Msuite/Utils/simple_run_checkm.py:
```diff
@@ -0,0 +1,24 @@
+        args.append(opt)
+        if val:
+            args.append(str(val))
+    proc = Popen(args, stdout=PIPE, stderr=STDOUT)
```
Added this in :heavy_check_mark:

jayrbolton commented:
@Tianhao-Gu 
- Bumped version
- Updated spec file to match current version of the lineage_wf params. Made the return structure empty. Added the log_path into params.

jayrbolton commented:
Thanks :pizza:
